network/wsNetwork.go

```diff
@@ -2089,6 +2148,11 @@ func (wn *WebsocketNetwork) tryConnect(addr, gossipAddr string) {
 	for _, supportedProtocolVersion := range wn.supportedProtocolVersions {
 		requestHeader.Add(ProtocolAcceptVersionHeader, supportedProtocolVersion)
 	}
+
+	idChal := NewIdentityChallenge((crypto.PublicKey)(wn.identityKeys.SignatureVerifier))
```

Comment:
aside from test code this is only used here. two ideas
1. make it take the type that wn.identityKeys.SignatureVerifier is so that we don't have to cast
2. combine 'New()' and 'SignAndEncodeB64()' into one function that takes wn.identityKeys